reference/basic-architecture.md

```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
-| *nats-config* | ConfigMap | NATS cluster configuration data |
-| etcd | Pod | Hosts etcd Server container | Single |
-| *mayastor-etcd* | Service | Exposes NATS message bus end point | Single |
-| *mayastor-etcd-headless* | Service | Exposes NATS message bus end point | Single |
-| **Data Plane**|
-| mayastor| Pod | Hosts Mayastor I/O engine| User-selected nodes |
-| **k8s Resource Types**  |
-| mayastorpools | CRD | Declares a Mayastor pool's desired state and reflects its current state | User-defined, one or many |
+| agent-core | Deployment | Principle control plane actor | Single |
+| csi-controller | Deployment | Hosts Mayastor's CSI controller implementation and CSI provisioner side car| Single |
+| api-rest | Pod | Hosts the public API REST server | Single |
+| *api-rest*  | Service | Exposes the REST API server via NodePort |
+| operator-diskpool | Deployment | Hosts DiskPool operator | Single |
+| csi-node| DaemonSet | Hosts CSI Driver node plugin containers | All worker nodes |
+| etcd | StatefulSet | Hosts etcd Server container | Configurable(<i>Recommended: Three replicas)</i> |
+| etcd | Service | Exposes etcd DB endpoint | Single |
+| etcd-headless | Service | Exposes etcd DB endpoint | Single |
+| io-engine| DaemonSet | Hosts Mayastor I/O engine| User-selected nodes |
+| DiskPool | CRD | Declares a DiskPool's desired state and reflects its current state | User-defined, one or many |
+| **Additional components**  |
+| metrics-exporter-pool | Sidecar container (within io-engine DaemonSet)| Exports pool related metrics in Prometheus format | All worker nodes |
+| pool-metrics-exporter | Service| Exposes exporter API endpoint to Prometheus | Single |
+| promtail | DaemonSet| Scrapes logs of Mayastor-specific pods and exports them to Loki| All worker nodes |
+| loki | StatefulSet| Stores the historical logs exported by promtail pods | Single |
+| loki | Service| Exposes the Loki API endpoint via ClusterIP | Single |
 
 ## Component Roles
 
-### Control Plane
+### io-engine
 
-A microservices patterned control plane, centered around a core agent with a publically exposed RESTful API.  This is extended by a dedicated operator responsible for managing the life cycle of "Mayastor Pools" (an abstraction for devices supplying the cluster with persistent backing storage), and a CSI compliant external provisioner (controller).  The source for the control plane components is located in its [own repository](https://github.com/mayadata-io/mayastor-control-plane)
+The io-engine pods encapsulate Mayastor containers, which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
+The instance of the Mayastor running inside the container performs four major classes of functions:
+- Presents a gRPC interface to the components which allow it to orchestrate the creation, configuration, and deletion of Mayastor-managed objects hosted by that instance.
```

Comment:
This can be the 4th point and be modified as "Provides a gRPC interface service to orchestrate the creation, deletion and management of the above objects, hosted on that node."